### PR TITLE
🌱 Use Kubernetes 1.23 in: quickstart, CAPD, e2e tests + update doc

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Partially implements #5793

Related: https://github.com/kubernetes/test-infra/pull/24554

### CODE REVIEW

**Comment by sbueringer** [comment]
/hold
until Kubernetes 1.23 is released (probably tomorrow)

File: docs/book/src/user/quick-start.md [hunk 1]
```diff
@@ -731,7 +731,7 @@ Calico is used here as an example.
 
 ```bash
 kubectl --kubeconfig=./capi-quickstart.kubeconfig \
-  apply -f https://docs.projectcalico.org/v3.20/manifests/calico.yaml
+  apply -f https://docs.projectcalico.org/v3.21/manifests/calico.yaml
```
Comment (sbueringer) [inline docs/book/src/user/quick-start.md:734]: NOTE: I've verified the quickstart locally with Kubernetes 1.23 and Calico v3.21

**Comment by sbueringer** [comment]
/test pull-cluster-api-e2e-main 
GitHub flake

**Comment by sbueringer** [comment]
/test pull-cluster-api-e2e-full-main
Another GitHub flake

**Review by fabriziopandini — COMMENTED** [review]
Thanks for taking over this effort!
/lgtm
/approve

**Pull request merged**